NGitLab/Impl/HttpRequestor.cs

```diff
@@ -162,14 +163,17 @@ public override async IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken c
                     using var streamReader = new StreamReader(stream);
                     var responseText = await streamReader.ReadToEndAsync().ConfigureAwait(false);
                     var deserialized = SimpleJson.DeserializeObject<T[]>(responseText);
-                    foreach (var item in deserialized)
-                        yield return item;
+                    buffer.AddRange(deserialized);
```

Comment:
By using a list, you cannot stream the response. You'll enumerate all pages before returning the first result to the caller. 

I think you should try to first change the scope of the response / stream by using explicit using block (`using (...) {}`), so you ensure the response and stream are disposed before returning the item.